OpenRA.Mods.Common/Traits/Mobile.cs

```diff
@@ -76,13 +76,20 @@ public override void RulesetLoaded(Ruleset rules, ActorInfo ai)
 
 		public int GetInitialFacing() { return InitialFacing; }
 
+		// initialized and used by CanEnterCell
+		Locomotor locomotor;
+
 		public bool CanEnterCell(World world, Actor self, CPos cell, Actor ignoreActor = null, bool checkTransientActors = true)
 		{
-			if (LocomotorInfo.MovementCostForCell(world, cell) == int.MaxValue)
+			if (locomotor == null)
```

Comment:
What do you think about adding a comment like `// PERF: Avoid repeated trait queries on the hot path`?